common/src/main/java/com/mx/common/accessors/UpstreamSystemUnavailable.java

```diff
@@ -0,0 +1,50 @@
+package com.mx.common.accessors;
+
+import com.mx.common.exception.PathRequestException;
+
+/**
+ * Thrown when an upstream request results in an unknown error or the upstream service is determined to be offline.
+ *
+ * <p>This can be used as a catch-all exception for situations where something went bad upstream,
+ * usually resulting in an unexpected status code, and we couldn't figure out why. This is close relative to
+ * {@link ResponsePayloadException}. The difference is that {@link ResponsePayloadException} indicates that we were
+ * unable to process a response that was expected to be good. This exception is a bit more general and used when
+ * we can determine that the response is not right.
+ *
+ * <p>Consider the following code example. In this example, we are attempting to determine why the response did not have
+ * a 200 OK status and defaulting to   Keep in mind, this is upstream-system-specific. Determining what happened, may not
+ * be possible, in which case, we do our best.
+ *
+ * <pre>{@code
+ * if (response.getStatus() != HttpStatus.OK) {
+ *   String code = extractErrorCodeFromResponseBody(response.getBody());
+ *
+ *   if ("validation".equals(code)) {
+ *     thrown new RequestValidationException("Invalid request", "The request was not valid. Please review and try again. If this error occurs again, please call customer service.")
+ *   } else if ("maintenance".equals(code)) {
+ *     thrown new UpstreamSystemUnavailable("Bank system maintenance", "Bank systems are down for maintenance. Please try again later.")
+ *   } else if ("internal".equals(code)) {
+ *     thrown new UpstreamSystemUnavailable("Bank system error", "Bank system failed to process request. Please try again later.")
+ *   } else { // default
+ *     thrown new UpstreamSystemUnavailable("Bank system unknown state", "Bank system failed to process request. Please try again later.")
+ *   }
+ * }
+ * }</pre>
+ *
+ * <p>See {@link PathRequestException} for usage details
+ */
+public class UpstreamSystemUnavailable extends AccessorSystemException {
+  public UpstreamSystemUnavailable(String message, String userMessage) {
+    super(message);
+    setReport(false);
+    setStatus(PathResponseStatus.UPSTREAM_SERVICE_UNAVAILABLE);
```

Comment:
Yesterday I was listening to a conversation with @Joe Hansen about this, specifically 531s, and he was pretty adamant that a 531 not just be ‘upstream unavailable’, but specifically be ‘maintenance mode’. We may want to follow up with him and/or synchronicity to make sure we’re all on the same page.

Reply:
531 is currently used to to convey for any unknown upstream error or upstream unavailable-ness. https://gitlab.mx.com/mx/synchronicity/-/blob/master/app/controllers/concerns/rescue_from_mdx_errors.rb#L7

We use it to indicate maintenance mode as well. We have limited buckets to throw errors into. UPSTREAM_SERVICE_UNAVAILABLE is a pretty big bucket and we can't limit its use to only one situation.

We can consider enhancing MDX to include additional information about the UPSTREAM_SERVICE_UNAVAILABLE. Synchronicity currently includes a user_message if the reason is that the service is in maintenance mode. We could use a codified indicator as well.

@JAHansens , Thoughts?

Reply:
After a long conversation with Matt, it turns out that the App has been misinterpreting this since it was created 9 years ago. we have some plans to fix the app side to bring it to the same definition as Sync.